ADD the URL Embed Text Filter

cs-shadow opened the pull request:
Working on #1 to make the tests work.

prateekmehta committed:
ADD the URL Embed Text Filter

prateekmehta committed:
remove composer dependecy.

prateekmehta committed:
create UrlEmbedFilterTest

prateekmehta committed:
Merge remote-tracking branch 'upstream2/8.x-1.x' into process_text

prateekmehta committed:
change variable name from testUrl to sample_urlOF

cs-shadow committed:
Fix composer update on Travis and tests

cs-shadow commented:
The test fails because Embed is returning error status 429. Tests should pass locally.

@prateekmehta  @davereid  Would appreciate if you both can test it locally before we merge this since the tests are failing on Travis atm.

davereid commented on src/Plugin/Filter/UrlEmbedFilter.php:
```diff
@@ -0,0 +1,102 @@
+   * @param string $content
+   *   The text or HTML that will replace the contents of $node.
+   */
+  protected function setDomNodeContent(\DOMNode $node, $content) {
```
Noting that as soon as possible we should add an Embed module dependency and re-use the function from DomHelperTrait

cs-shadow replied:
Created a followup just so that this is not overlooked: https://www.drupal.org/node/2547105

davereid committed:
Update UrlEmbedFilter.php

davereid committed:
Update UrlEmbedFilter.php

davereid commented:
Looks like we currently have a merge conflict that needs to be resolved. Ideally I'd like to get this committed once we also have the internal module we can use for testing embedding.

cs-shadow committed:
Merge 8.x-1.x and fix conflicts

cs-shadow commented:
@davereid Fixed the conflicts and ran the tests again locally to ensure it still works. Although ideally we should wait for the test module but I suggest merging it now so that other issues (like ckeditor integration) are not blocked by it.

davereid commented:
Yeah, I just don't want to get too far into the CKEditor integration without having the other parts ready to go, like field formatters (and hence display plugins).

davereid commented on scripts/composer-update.sh:
```diff
@@ -0,0 +1,12 @@
+drupal_ti_ensure_drupal
+
+cd "$DRUPAL_TI_DRUPAL_DIR"
+cd core
```
This could be just written as `cd "$DRUPAL_TI_DRUPAL_DIR\core"` ?

davereid commented on scripts/composer-update.sh:
```diff
@@ -0,0 +1,12 @@
+cd "$DRUPAL_TI_DRUPAL_DIR"
+cd core
+composer drupal-rebuild
+composer update embed --prefer-source
```
Why is this necessary (just curious)?

cs-shadow replied:
Sometimes travis fails to download packages from Github, due to API limit or whatever. This option helps avoid that. One issue which I referenced: https://github.com/composer/composer/issues/1314

davereid commented on src/Plugin/Filter/UrlEmbedFilter.php:
```diff
@@ -0,0 +1,103 @@
+ * @Filter(
+ *   id = "url_embed",
+ *   title = @Translation("Display embedded URLs"),
+ *   description = @Translation("Embeds URLs using data attributes: url, source, oembed provider and view-mode."),
```
References view-mode here, remove.

cs-shadow committed:
Use flickr url instead of youtube for testing

cs-shadow commented:
@davereid Addressed your comments and also replaced youtube url for testing with a fickr url which seems to work fine. And from what it seems flickr doesn't blacklist travis often so we should be good.

The tests are now passing on Travis but fail on PHP 7 for no apparent reason. It looks like something wrong with our custom script but I suggest to get this PR in so that @prateekmehta is not blocked by it.

davereid commented on src/Plugin/Filter/UrlEmbedFilter.php:
```diff
@@ -0,0 +1,103 @@
+ * @Filter(
+ *   id = "url_embed",
+ *   title = @Translation("Display embedded URLs"),
+ *   description = @Translation("Embeds URLs using data attribute: data-url."),
```
Should have noted earlier, the attribute is actually data-embed-url

davereid committed:
Update UrlEmbedFilter.php

cs-shadow commented:
@davereid Any idea whats up with PHP 7 build?

Berdir commented:
https://www.drupal.org/node/2547715 is up with PHP 7.

cs-shadow commented:
@Berdir thanks! That makes more sense, I thought I broke the build again right after I barely got it to work.
